extensions/resource-deployment/src/ui/deployAzureSQLDBWizard/pages/azureSettingsPage.ts

```diff
@@ -321,12 +309,7 @@ export class AzureSettingsPage extends BasePage {
 		let response = await this.wizard.getRequest(url);
 		if (response.data.value.length === 0) {
 			this._serverGroupDropdown.updateProperties({
-				values: [
-					{
-						displayName: localize('deployAzureSQLDB.NoServerLabel', "No servers found"),
-						name: ''
-					}
-				],
+				values: [],
```

Comment:
In general I would say it's useful to have an indication that there's no values available. Why did you remove this? Does the user get notified in some other way that this happened?

Reply:
Currently the dropdown class itself does not support validationerrormessages, I will need to implement it for the class but that will take time.

Reply:
I'm not really sure the validation message is appropriate here anyways - I was mostly referring to removing the default option that's displayed when there's no other valid options.

Reply:
@ranasaria and @aasimkhan30 told me that they would prefer I should not have a mock value in the server dropdown, in Aasim's azure settings page, the resource group dropdown is also programmed to display blank if no values are found. However there is no option for a placeholder value for dropdowns currently, its either empty without a message or it will have to have a mock value in it, (which requires additional validation)

Reply:
I commented above. The basic idea is if or not the subscription dropdown being 'empty' is a fatal error. If fatal error, then in my opinion we should pop the error in red on top of the wizard and make sure that the wizard does not continue until the user takes appropriate corrective action so that valid subscription information can get fetched. If this is optional then I would agree with Charles on displaying a value to indicate that no subscriptions were found.

Reply:
Considering that we already have required fields with text validation I don't necessarily agree that just showing the error message is correct for your scenario. In my mind there's two things that I think should happen there :

1. Likely some backend error happened when fetching subs (or whatever the option items are) This should probably be displayed to the user using the error message/error toast.
1. But then also we still want to indicate in the dropdown that it did finish fetching the items but was unable to find any valid ones. This should make it so the dialog is unable to continue (assuming the field is required) and highlight the box in red to indicate a valid value must be chosen. This is for accessibility reasons so that someone tabbing over the box knows the state but is also just a useful thing to have for even normal users since it's not always clear when things are done loading for async operations like this.

These events are not necessarily connected. For example you could have an error fetching some options (for example RG's in one sub, but the rest populate correctly). And you could also have it return without an error but have no items. Both cases we need to ensure are handled and displayed correctly so the user isn't left wondering whether items are still being loaded or what errors occurred.